rsocket-tck-drivers/src/test/java/io/rsocket/tckdrivers/TckTest.java

```diff
@@ -79,19 +87,50 @@ public void TckTestRunner() {
 
     assertNotNull("JavaClientDriver is not defined", jd);
     try {
-      jd.runTest(this.test.subList(1, this.test.size()), this.name);
+      jd.runTest(this.tckTest.test.subList(1, this.tckTest.test.size()), this.tckTest.name);
     } catch (Exception e) {
       e.printStackTrace();
     }
   }
 
   /**
+   * A function that parses the file and extract the individual tests
+   *
+   * @param file: file to read as input.
+   * @return a list of individual tests. Each individual test is also a list of String.
+   */
+  private static List<List<String>> ExtractTests(File file) throws Exception {
```

Comment:
If you change this method to return List<TckIndividualTest> then we are moving towards a model where the input could be a library and calling a function instead of assuming there is a file to read from.